### ISSUE TITLE
csi-cephfsplugin-provisioner and csi-rbdplugin-provisioner pods always crash in openshift, can't be ready.

### ISSUE THREAD

**Issue by huangxuelun** [rook/rook#9001]
![image](https://user-images.githubusercontent.com/3275246/137827598-3b5623cf-be2c-4ab3-bf4b-304bc69ce34f.png)
just follow the installation document [https://rook.io/docs/rook/v1.7/ceph-openshift.html](url)
**os:**
Linux bastion 4.18.0-305.3.1.el8.x86_64 #1 SMP Tue Jun 1 16:14:33 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
**rook version** is 1.7.5
**openshit info:**
Client Version: 4.6.27
Server Version: 4.6.27
Kubernetes Version: v1.19.0+d856161

log:

**from csi-cephfsplugin-provisioner:**
W1019 01:22:10.724878       1 connection.go:173] Still connecting to unix:///csi/csi-provisioner.sock
W1019 01:22:20.721970       1 connection.go:173] Still connecting to unix:///csi/csi-provisioner.sock
W1019 01:22:30.719687       1 connection.go:173] Still connecting to unix:///csi/csi-provisioner.sock
W1019 01:22:40.721680       1 connection.go:173] Still connecting to unix:///csi/csi-provisioner.sock

**from csi-rbdplugin-provisioner:**
W1019 01:24:04.481335       1 connection.go:173] Still connecting to unix:///csi/csi-provisioner.sock
W1019 01:24:14.481741       1 connection.go:173] Still connecting to unix:///csi/csi-provisioner.sock
W1019 01:24:24.482254       1 connection.go:173] Still connecting to unix:///csi/csi-provisioner.sock
W1019 01:24:34.481673       1 connection.go:173] Still connecting to unix:///csi/csi-provisioner.sock

**from kubelet:**
Warning: The unit file, source configuration file or drop-ins of kubelet.service changed on disk. Run 'systemctl daemon-reload' to reload units.
● kubelet.service - Kubernetes Kubelet
   Loaded: loaded (/etc/systemd/system/kubelet.service; enabled; vendor preset: enabled)
  Drop-In: /etc/systemd/system/kubelet.service.d
           └─10-mco-default-env.conf, 20-nodenet.conf
   Active: active (running) since Wed 2021-10-13 07:00:12 UTC; 5 days ago
  Process: 2309 ExecStartPre=/bin/rm -f /var/lib/kubelet/cpu_manager_state (code=exited, status=0/SUCCESS)
  Process: 2307 ExecStartPre=/bin/mkdir --parents /etc/kubernetes/manifests (code=exited, status=0/SUCCESS)
 Main PID: 2312 (kubelet)
    Tasks: 43 (limit: 308489)
   Memory: 406.0M
      CPU: 3d 5h 33min 14.962s
   CGroup: /system.slice/kubelet.service
           └─2312 kubelet --config=/etc/kubernetes/kubelet.conf --bootstrap-kubeconfig=/etc/kubernetes/kubeconfig --kubeconfig=/var/lib/kubelet/kubeconfig --container-runtime=remote --container-runtime-endpoint=/var/run/crio/crio.sock --runtime-cgroups=/system.slice/crio.service --node-labels=node-role.kubernetes.io/worker,node.openshift.io/os_id=rhcos --node-ip=172.16.235.46 --minimum-container-ttl-duration=6m0s --volume-plugin-dir=/etc/kubernetes/kubelet-plugins/volume/exec --cloud-provider= --pod-infra-container-image=quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:aa6410430d05d37ac258c34c2279653595ea82dcaabab5ad1bd5c1eb8db14c39 --v=3

Oct 19 01:26:09 worker01.ocp.example.com hyperkube[2312]: I1019 01:26:09.279912    2312 secret.go:183] Setting up volume audit-server-certs for pod 942e5291-46c7-4f9f-8473-d171e9895274 at /var/lib/kubelet/pods/942e5291-46c7-4f9f-8473-d171e9895274/volumes/kubernetes.io~secret/audit-server-certs
Oct 19 01:26:09 worker01.ocp.example.com hyperkube[2312]: I1019 01:26:09.280907    2312 secret.go:183] Setting up volume cluster-certs for pod 942e5291-46c7-4f9f-8473-d171e9895274 at /var/lib/kubelet/pods/942e5291-46c7-4f9f-8473-d171e9895274/volumes/kubernetes.io~secret/cluster-certs
Oct 19 01:26:09 worker01.ocp.example.com hyperkube[2312]: I1019 01:26:09.280922    2312 configmap.go:188] Setting up volume audit-ingest for pod 942e5291-46c7-4f9f-8473-d171e9895274 at /var/lib/kubelet/pods/942e5291-46c7-4f9f-8473-d171e9895274/volumes/kubernetes.io~configmap/audit-ingest
Oct 19 01:26:09 worker01.ocp.example.com hyperkube[2312]: I1019 01:26:09.280929    2312 secret.go:207] Received secret ibm-common-services/audit-server-certs containing (0) pieces of data, 0 total bytes
Oct 19 01:26:09 worker01.ocp.example.com hyperkube[2312]: I1019 01:26:09.281063    2312 secret.go:207] Received secret ibm-common-services/platform-api-secret containing (3) pieces of data, 4031 total bytes
Oct 19 01:26:09 worker01.ocp.example.com hyperkube[2312]: I1019 01:26:09.281122    2312 configmap.go:212] Received configMap ibm-common-services/audit-logging-fluentd-ds-http-ingesturl containing (0) pieces of data, 0 total bytes
Oct 19 01:26:09 worker01.ocp.example.com hyperkube[2312]: I1019 01:26:09.281534    2312 secret.go:183] Setting up volume ibm-platform-api-operand-token-9jdgl for pod 942e5291-46c7-4f9f-8473-d171e9895274 at /var/lib/kubelet/pods/942e5291-46c7-4f9f-8473-d171e9895274/volumes/kubernetes.io~secret/ibm-platform-api-operand-token-9jdgl
Oct 19 01:26:09 worker01.ocp.example.com hyperkube[2312]: I1019 01:26:09.281685    2312 secret.go:207] Received secret ibm-common-services/ibm-platform-api-operand-token-9jdgl containing (4) pieces of data, 16672 total bytes
Oct 19 01:26:09 worker01.ocp.example.com hyperkube[2312]: I1019 01:26:09.567417    2312 prober.go:126] Readiness probe for "dns-default-whzgw_openshift-dns(ecaa9c58-8e50-475a-96ed-59bbad53ee79):dns" succeeded
Oct 19 01:26:09 worker01.ocp.example.com hyperkube[2312]: I1019 01:26:09.588429    2312 prober.go:126] Readiness probe for "certified-operators-vhq7t_openshift-marketplace(9932f859-2b2b-4fb5-8ff4-7713056f74de)

**Comment by Madhu-1** [comment]
@huangxuelun please provide logs of all containers in  provisioner pods

**Comment by huangxuelun** [comment]
apiVersion: v1
kind: Pod
metadata:
  annotations:
    k8s.v1.cni.cncf.io/network-status: |-
      [{
          "name": "",
          "interface": "eth0",
          "ips": [
              "10.131.1.93"
          ],
          "default": true,
          "dns": {}
      }]
    k8s.v1.cni.cncf.io/networks-status: |-
      [{
          "name": "",
          "interface": "eth0",
          "ips": [
              "10.131.1.93"
          ],
          "default": true,
          "dns": {}
      }]
    openshift.io/scc: restricted
  creationTimestamp: "2021-10-19T04:30:32Z"
  generateName: csi-cephfsplugin-provisioner-54fdf994d6-
  labels:
    app: csi-cephfsplugin-provisioner
    contains: csi-cephfsplugin-metrics
    pod-template-hash: 54fdf994d6
  managedFields:
  - apiVersion: v1
    fieldsType: FieldsV1
    fieldsV1:
      f:metadata:
        f:generateName: {}
        f:labels:
          .: {}
          f:app: {}
          f:contains: {}
          f:pod-template-hash: {}
        f:ownerReferences:
          .: {}
          k:{"uid":"c35fd60c-eca4-41fd-8736-f283ed1bf8fb"}:
            .: {}
            f:apiVersion: {}
            f:blockOwnerDeletion: {}
            f:controller: {}
            f:kind: {}
            f:name: {}
            f:uid: {}
      f:spec:
        f:affinity:
          .: {}
          f:nodeAffinity: {}
          f:podAntiAffinity:
            .: {}
            f:requiredDuringSchedulingIgnoredDuringExecution: {}
        f:containers:
          k:{"name":"csi-attacher"}:
            .: {}
            f:args: {}
            f:env:
              .: {}
              k:{"name":"ADDRESS"}:
                .: {}
                f:name: {}
                f:value: {}
            f:image: {}
            f:imagePullPolicy: {}
            f:name: {}
            f:resources: {}
            f:terminationMessagePath: {}
            f:terminationMessagePolicy: {}
            f:volumeMounts:
              .: {}
              k:{"mountPath":"/csi"}:
                .: {}
                f:mountPath: {}
                f:name: {}
          k:{"name":"csi-cephfsplugin"}:
            .: {}
            f:args: {}
            f:env:
              .: {}
              k:{"name":"CSI_ENDPOINT"}:
                .: {}
                f:name: {}
                f:value: {}
              k:{"name":"NODE_ID"}:
                .: {}
                f:name: {}
                f:valueFrom:
                  .: {}
                  f:fieldRef:
                    .: {}
                    f:apiVersion: {}
                    f:fieldPath: {}
              k:{"name":"POD_IP"}:
                .: {}
                f:name: {}
                f:valueFrom:
                  .: {}
                  f:fieldRef:
                    .: {}
                    f:apiVersion: {}
                    f:fieldPath: {}
              k:{"name":"POD_NAMESPACE"}:
                .: {}
                f:name: {}
                f:valueFrom:
                  .: {}
                  f:fieldRef:
                    .: {}
                    f:apiVersion: {}
                    f:fieldPath: {}
            f:image: {}
            f:imagePullPolicy: {}
            f:name: {}
            f:resources: {}
            f:terminationMessagePath: {}
            f:terminationMessagePolicy: {}
            f:volumeMounts:
              .: {}
              k:{"mountPath":"/csi"}:
                .: {}
                f:mountPath: {}
                f:name: {}
              k:{"mountPath":"/dev"}:
                .: {}
                f:mountPath: {}
                f:name: {}
              k:{"mountPath":"/etc/ceph-csi-config/"}:
                .: {}
                f:mountPath: {}
                f:name: {}
              k:{"mountPath":"/lib/modules"}:
                .: {}
                f:mountPath: {}
                f:name: {}
                f:readOnly: {}
              k:{"mountPath":"/sys"}:
                .: {}
                f:mountPath: {}
                f:name: {}
              k:{"mountPath":"/tmp/csi/keys"}:
                .: {}
                f:mountPath: {}
                f:name: {}
          k:{"name":"csi-provisioner"}:
            .: {}
            f:args: {}
            f:env:
              .: {}
              k:{"name":"ADDRESS"}:
                .: {}
                f:name: {}
                f:value: {}
            f:image: {}
            f:imagePullPolicy: {}
            f:name: {}
            f:resources: {}
            f:terminationMessagePath: {}
            f:terminationMessagePolicy: {}
            f:volumeMounts:
              .: {}
              k:{"mountPath":"/csi"}:
                .: {}
                f:mountPath: {}
                f:name: {}
          k:{"name":"csi-resizer"}:
            .: {}
            f:args: {}
            f:env:
              .: {}
              k:{"name":"ADDRESS"}:
                .: {}
                f:name: {}
                f:value: {}
            f:image: {}
            f:imagePullPolicy: {}
            f:name: {}
            f:resources: {}
            f:terminationMessagePath: {}
            f:terminationMessagePolicy: {}
            f:volumeMounts:
              .: {}
              k:{"mountPath":"/csi"}:
                .: {}
                f:mountPath: {}
                f:name: {}
          k:{"name":"csi-snapshotter"}:
            .: {}
            f:args: {}
            f:env:
              .: {}
              k:{"name":"ADDRESS"}:
                .: {}
                f:name: {}
                f:value: {}
            f:image: {}
            f:imagePullPolicy: {}
            f:name: {}
            f:resources: {}
            f:terminationMessagePath: {}
            f:terminationMessagePolicy: {}
            f:volumeMounts:
              .: {}
              k:{"mountPath":"/csi"}:
                .: {}
                f:mountPath: {}
                f:name: {}
          k:{"name":"liveness-prometheus"}:
            .: {}
            f:args: {}
            f:env:
              .: {}
              k:{"name":"CSI_ENDPOINT"}:
                .: {}
                f:name: {}
                f:value: {}
              k:{"name":"POD_IP"}:
                .: {}
                f:name: {}
                f:valueFrom:
                  .: {}
                  f:fieldRef:
                    .: {}
                    f:apiVersion: {}
                    f:fieldPath: {}
            f:image: {}
            f:imagePullPolicy: {}
            f:name: {}
            f:resources: {}
            f:terminationMessagePath: {}
            f:terminationMessagePolicy: {}
            f:volumeMounts:
              .: {}
              k:{"mountPath":"/csi"}:
                .: {}
                f:mountPath: {}
                f:name: {}
        f:dnsPolicy: {}
        f:enableServiceLinks: {}
        f:restartPolicy: {}
        f:schedulerName: {}
        f:securityContext: {}
        f:serviceAccount: {}
        f:serviceAccountName: {}
        f:terminationGracePeriodSeconds: {}
        f:volumes:
          .: {}
          k:{"name":"ceph-csi-config"}:
            .: {}
            f:configMap:
              .: {}
              f:defaultMode: {}
              f:items: {}
              f:name: {}
            f:name: {}
          k:{"name":"host-dev"}:
            .: {}
            f:hostPath:
              .: {}
              f:path: {}
              f:type: {}
            f:name: {}
          k:{"name":"host-sys"}:
            .: {}
            f:hostPath:
              .: {}
              f:path: {}
              f:type: {}
            f:name: {}
          k:{"name":"keys-tmp-dir"}:
            .: {}
            f:emptyDir:
              .: {}
              f:medium: {}
            f:name: {}
          k:{"name":"lib-modules"}:
            .: {}
            f:hostPath:
              .: {}
              f:path: {}
              f:type: {}
            f:name: {}
          k:{"name":"socket-dir"}:
            .: {}
            f:emptyDir:
              .: {}
              f:medium: {}
            f:name: {}
    manager: kube-controller-manager
    operation: Update
    time: "2021-10-19T04:30:32Z"
  - apiVersion: v1
    fieldsType: FieldsV1
    fieldsV1:
      f:metadata:
        f:annotations:
          f:k8s.v1.cni.cncf.io/network-status: {}
          f:k8s.v1.cni.cncf.io/networks-status: {}
    manager: multus
    operation: Update
    time: "2021-10-19T04:30:35Z"
  - apiVersion: v1
    fieldsType: FieldsV1
    fieldsV1:
      f:status:
        f:conditions:
          k:{"type":"ContainersReady"}:
            .: {}
            f:lastProbeTime: {}
            f:lastTransitionTime: {}
            f:message: {}
            f:reason: {}
            f:status: {}
            f:type: {}
          k:{"type":"Initialized"}:
            .: {}
            f:lastProbeTime: {}
            f:lastTransitionTime: {}
            f:status: {}
            f:type: {}
          k:{"type":"Ready"}:
            .: {}
            f:lastProbeTime: {}
            f:lastTransitionTime: {}
            f:message: {}
            f:reason: {}
            f:status: {}
            f:type: {}
        f:containerStatuses: {}
        f:hostIP: {}
        f:phase: {}
        f:podIP: {}
        f:podIPs:
          .: {}
          k:{"ip":"10.131.1.93"}:
            .: {}
            f:ip: {}
        f:startTime: {}
    manager: kubelet
    operation: Update
    time: "2021-10-19T04:34:02Z"
  name: csi-cephfsplugin-provisioner-54fdf994d6-cw4rr
  namespace: rook-ceph
  ownerReferences:
  - apiVersion: apps/v1
    blockOwnerDeletion: true
    controller: true
    kind: ReplicaSet
    name: csi-cephfsplugin-provisioner-54fdf994d6
    uid: c35fd60c-eca4-41fd-8736-f283ed1bf8fb
  resourceVersion: "15926575"
  selfLink: /api/v1/namespaces/rook-ceph/pods/csi-cephfsplugin-provisioner-54fdf994d6-cw4rr
  uid: a3a3c78f-fbdc-43e7-b383-84c4e104afbf
spec:
  affinity:
    nodeAffinity: {}
    podAntiAffinity:
      requiredDuringSchedulingIgnoredDuringExecution:
      - labelSelector:
          matchExpressions:
          - key: app
            operator: In
            values:
            - csi-cephfsplugin-provisioner
        topologyKey: kubernetes.io/hostname
  containers:
  - args:
    - --v=0
    - --csi-address=$(ADDRESS)
    - --leader-election=true
    - --timeout=150s
    - --leader-election-namespace=rook-ceph
    env:
    - name: ADDRESS
      value: /csi/csi-provisioner.sock
    image: k8s.gcr.io/sig-storage/csi-attacher:v3.3.0
    imagePullPolicy: IfNotPresent
    name: csi-attacher
    resources: {}
    securityContext:
      capabilities:
        drop:
        - KILL
        - MKNOD
        - SETGID
        - SETUID
      runAsUser: 1000630000
    terminationMessagePath: /dev/termination-log
    terminationMessagePolicy: File
    volumeMounts:
    - mountPath: /csi
      name: socket-dir
    - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
      name: rook-csi-cephfs-provisioner-sa-token-drg5s
      readOnly: true
  - args:
    - --csi-address=$(ADDRESS)
    - --v=0
    - --timeout=150s
    - --leader-election=true
    - --leader-election-namespace=rook-ceph
    env:
    - name: ADDRESS
      value: unix:///csi/csi-provisioner.sock
    image: k8s.gcr.io/sig-storage/csi-snapshotter:v4.2.0
    imagePullPolicy: IfNotPresent
    name: csi-snapshotter
    resources: {}
    securityContext:
      capabilities:
        drop:
        - KILL
        - MKNOD
        - SETGID
        - SETUID
      runAsUser: 1000630000
    terminationMessagePath: /dev/termination-log
    terminationMessagePolicy: File
    volumeMounts:
    - mountPath: /csi
      name: socket-dir
    - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
      name: rook-csi-cephfs-provisioner-sa-token-drg5s
      readOnly: true
  - args:
    - --csi-address=$(ADDRESS)
    - --v=0
    - --timeout=150s
    - --leader-election
    - --leader-election-namespace=rook-ceph
    - --handle-volume-inuse-error=false
    env:
    - name: ADDRESS
      value: unix:///csi/csi-provisioner.sock
    image: k8s.gcr.io/sig-storage/csi-resizer:v1.3.0
    imagePullPolicy: IfNotPresent
    name: csi-resizer
    resources: {}
    securityContext:
      capabilities:
        drop:
        - KILL
        - MKNOD
        - SETGID
        - SETUID
      runAsUser: 1000630000
    terminationMessagePath: /dev/termination-log
    terminationMessagePolicy: File
    volumeMounts:
    - mountPath: /csi
      name: socket-dir
    - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
      name: rook-csi-cephfs-provisioner-sa-token-drg5s
      readOnly: true
  - args:
    - --csi-address=$(ADDRESS)
    - --v=0
    - --timeout=150s
    - --retry-interval-start=500ms
    - --leader-election=true
    - --leader-election-namespace=rook-ceph
    env:
    - name: ADDRESS
      value: unix:///csi/csi-provisioner.sock
    image: k8s.gcr.io/sig-storage/csi-provisioner:v3.0.0
    imagePullPolicy: IfNotPresent
    name: csi-provisioner
    resources: {}
    securityContext:
      capabilities:
        drop:
        - KILL
        - MKNOD
        - SETGID
        - SETUID
      runAsUser: 1000630000
    terminationMessagePath: /dev/termination-log
    terminationMessagePolicy: File
    volumeMounts:
    - mountPath: /csi
      name: socket-dir
    - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
      name: rook-csi-cephfs-provisioner-sa-token-drg5s
      readOnly: true
  - args:
    - --nodeid=$(NODE_ID)
    - --type=cephfs
    - --endpoint=$(CSI_ENDPOINT)
    - --v=0
    - --controllerserver=true
    - --drivername=rook-ceph.cephfs.csi.ceph.com
    - --pidlimit=-1
    - --metricsport=9091
    - --forcecephkernelclient=true
    - --metricspath=/metrics
    - --enablegrpcmetrics=false
    env:
    - name: POD_IP
      valueFrom:
        fieldRef:
          apiVersion: v1
          fieldPath: status.podIP
    - name: NODE_ID
      valueFrom:
        fieldRef:
          apiVersion: v1
          fieldPath: spec.nodeName
    - name: POD_NAMESPACE
      valueFrom:
        fieldRef:
          apiVersion: v1
          fieldPath: metadata.namespace
    - name: CSI_ENDPOINT
      value: unix:///csi/csi-provisioner.sock
    image: quay.io/cephcsi/cephcsi:v3.4.0
    imagePullPolicy: IfNotPresent
    name: csi-cephfsplugin
    resources: {}
    securityContext:
      capabilities:
        drop:
        - KILL
        - MKNOD
        - SETGID
        - SETUID
      runAsUser: 1000630000
    terminationMessagePath: /dev/termination-log
    terminationMessagePolicy: File
    volumeMounts:
    - mountPath: /csi
      name: socket-dir
    - mountPath: /sys
      name: host-sys
    - mountPath: /lib/modules
      name: lib-modules
      readOnly: true
    - mountPath: /dev
      name: host-dev
    - mountPath: /etc/ceph-csi-config/
      name: ceph-csi-config
    - mountPath: /tmp/csi/keys
      name: keys-tmp-dir
    - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
      name: rook-csi-cephfs-provisioner-sa-token-drg5s
      readOnly: true
  - args:
    - --type=liveness
    - --endpoint=$(CSI_ENDPOINT)
    - --metricsport=9081
    - --metricspath=/metrics
    - --polltime=60s
    - --timeout=3s
    env:
    - name: CSI_ENDPOINT
      value: unix:///csi/csi-provisioner.sock
    - name: POD_IP
      valueFrom:
        fieldRef:
          apiVersion: v1
          fieldPath: status.podIP
    image: quay.io/cephcsi/cephcsi:v3.4.0
    imagePullPolicy: IfNotPresent
    name: liveness-prometheus
    resources: {}
    securityContext:
      capabilities:
        drop:
        - KILL
        - MKNOD
        - SETGID
        - SETUID
      runAsUser: 1000630000
    terminationMessagePath: /dev/termination-log
    terminationMessagePolicy: File
    volumeMounts:
    - mountPath: /csi
      name: socket-dir
    - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
      name: rook-csi-cephfs-provisioner-sa-token-drg5s
      readOnly: true
  dnsPolicy: ClusterFirst
  enableServiceLinks: true
  imagePullSecrets:
  - name: rook-csi-cephfs-provisioner-sa-dockercfg-q5zpc
  nodeName: worker03.ocp.example.com
  preemptionPolicy: PreemptLowerPriority
  priority: 0
  restartPolicy: Always
  schedulerName: default-scheduler
  securityContext:
    seLinuxOptions:
      level: s0:c25,c15
  serviceAccount: rook-csi-cephfs-provisioner-sa
  serviceAccountName: rook-csi-cephfs-provisioner-sa
  terminationGracePeriodSeconds: 30
  tolerations:
  - effect: NoExecute
    key: node.kubernetes.io/not-ready
    operator: Exists
    tolerationSeconds: 300
  - effect: NoExecute
    key: node.kubernetes.io/unreachable
    operator: Exists
    tolerationSeconds: 300
  volumes:
  - emptyDir:
      medium: Memory
    name: socket-dir
  - hostPath:
      path: /sys
      type: ""
    name: host-sys
  - hostPath:
      path: /lib/modules
      type: ""
    name: lib-modules
  - hostPath:
      path: /dev
      type: ""
    name: host-dev
  - configMap:
      defaultMode: 420
      items:
      - key: csi-cluster-config-json
        path: config.json
      name: rook-ceph-csi-config
    name: ceph-csi-config
  - emptyDir:
      medium: Memory
    name: keys-tmp-dir
  - name: rook-csi-cephfs-provisioner-sa-token-drg5s
    secret:
      defaultMode: 420
      secretName: rook-csi-cephfs-provisioner-sa-token-drg5s
status:
  conditions:
  - lastProbeTime: null
    lastTransitionTime: "2021-10-19T04:30:33Z"
    status: "True"
    type: Initialized
  - lastProbeTime: null
    lastTransitionTime: "2021-10-19T04:30:43Z"
    message: 'containers with unready status: [csi-cephfsplugin]'
    reason: ContainersNotReady
    status: "False"
    type: Ready
  - lastProbeTime: null
    lastTransitionTime: "2021-10-19T04:30:43Z"
    message: 'containers with unready status: [csi-cephfsplugin]'
    reason: ContainersNotReady
    status: "False"
    type: ContainersReady
  - lastProbeTime: null
    lastTransitionTime: "2021-10-19T04:30:33Z"
    status: "True"
    type: PodScheduled
  containerStatuses:
  - containerID: cri-o://c1811c6a50d0fa7d9e1dd40a8196721dca6289cc2960dff3556e1c6beb261343
    image: k8s.gcr.io/sig-storage/csi-attacher:v3.3.0
    imageID: k8s.gcr.io/sig-storage/csi-attacher@sha256:a02b6fbefa54cada55bb3eeefc5a66778e67e01e2d6f22889ce7a5c937d70313
    lastState: {}
    name: csi-attacher
    ready: true
    restartCount: 0
    started: true
    state:
      running:
        startedAt: "2021-10-19T04:30:37Z"
  - containerID: cri-o://7a596c12bf40fcbb445146c30871ae24cfaafb82fa60132c7071f467fc735b5d
    image: quay.io/cephcsi/cephcsi:v3.4.0
    imageID: quay.io/cephcsi/cephcsi@sha256:19634b6ef9fc6df2902cf6ff0b3dbccc56a6663d0cbfd065da44ecd2f955d848
    lastState:
      terminated:
        containerID: cri-o://7a596c12bf40fcbb445146c30871ae24cfaafb82fa60132c7071f467fc735b5d
        exitCode: 255
        finishedAt: "2021-10-19T04:33:47Z"
        reason: Error
        startedAt: "2021-10-19T04:33:47Z"
    name: csi-cephfsplugin
    ready: false
    restartCount: 5
    started: false
    state:
      waiting:
        message: back-off 2m40s restarting failed container=csi-cephfsplugin pod=csi-cephfsplugin-provisioner-54fdf994d6-cw4rr_rook-ceph(a3a3c78f-fbdc-43e7-b383-84c4e104afbf)
        reason: CrashLoopBackOff
  - containerID: cri-o://0ea495f3b5e2d7412b07eab1cbd55918c304c08f04d613677ac8717e267bbd13
    image: k8s.gcr.io/sig-storage/csi-provisioner:v3.0.0
    imageID: k8s.gcr.io/sig-storage/csi-provisioner@sha256:64430cb4a367d127943688833fabbf43e1e65fd95596f9324700fc17c039fbdd
    lastState: {}
    name: csi-provisioner
    ready: true
    restartCount: 0
    started: true
    state:
      running:
        startedAt: "2021-10-19T04:30:39Z"
  - containerID: cri-o://7d995be150a68baabc1800bb7eca1612f3493bf6a4c3e5b108cbc7f5fd44c426
    image: k8s.gcr.io/sig-storage/csi-resizer:v1.3.0
    imageID: k8s.gcr.io/sig-storage/csi-resizer@sha256:d3e0985228924d7d909d82c747b09611060cae07c6a1a080e7133c2977813c53
    lastState: {}
    name: csi-resizer
    ready: true
    restartCount: 0
    started: true
    state:
      running:
        startedAt: "2021-10-19T04:30:38Z"
  - containerID: cri-o://04c6bfdb1fd1e4cd63def6795fba259260d4a7c6e4afe006cabee85ee88240b1
    image: k8s.gcr.io/sig-storage/csi-snapshotter:v4.2.0
    imageID: k8s.gcr.io/sig-storage/csi-snapshotter@sha256:a6ae16681cf2a7aa9274452643be3c7362907e176af9b7d21d231f40b1b541e3
    lastState: {}
    name: csi-snapshotter
    ready: true
    restartCount: 0
    started: true
    state:
      running:
        startedAt: "2021-10-19T04:30:37Z"
  - containerID: cri-o://50dcf1e4268a09905a40cf39cdad101484a6eba83ab51060380900f0f3e0358d
    image: quay.io/cephcsi/cephcsi:v3.4.0
    imageID: quay.io/cephcsi/cephcsi@sha256:19634b6ef9fc6df2902cf6ff0b3dbccc56a6663d0cbfd065da44ecd2f955d848
    lastState: {}
    name: liveness-prometheus
    ready: true
    restartCount: 0
    started: true
    state:
      running:
        startedAt: "2021-10-19T04:30:41Z"
  hostIP: 172.16.235.48
  phase: Running
  podIP: 10.131.1.93
  podIPs:
  - ip: 10.131.1.93
  qosClass: BestEffort
  startTime: "2021-10-19T04:30:33Z"

**Comment by Madhu-1** [comment]
>openshift.io/scc: restricted

this is the culprit, the SCC defined in the operator-openshift.yaml is https://github.com/rook/rook/blob/d2a03bb2b35fc617ac7ab21a34e4f2067d864310/cluster/examples/kubernetes/ceph/operator-openshift.yaml#L55 not sure why restricted is applied.

**Comment by huangxuelun** [comment]
I don't have any change on rook yaml files. look like openshift added it.  so what should I do?

**Comment by huangxuelun** [comment]
![image](https://user-images.githubusercontent.com/3275246/137845886-387d3177-7fdd-49de-9797-5c59058609f3.png)

**Comment by eclipsewebmaster** [comment]
We are seeing this as well on our test setup(with an external ceph cluster) after a node restart and then a cluster upgrade

okd 4.7.0-0.okd-2021-09-19-013247
k8s v1.20.7-1330+bbbc0799c02950-dirty
rook 1.7.8

And the csi-rbdplugin container is throwing the same error:

```
E1129 20:31:08.210964       1 cephcsi.go:189] Failed to set new PID limit to -1: open /sys/fs/cgroup/pids/kubepods.slice/kubepods-besteffort.slice/kubepods-besteffort-pod7dc0aed1_1e86_4206_8fc6_80dce94c6229.slice/crio-b832befc598be453db4ceb548ec728536441553d3d186995b88fe7d3a263cf31.scope/pids.max: permission denied
F1129 20:31:08.211363       1 driver.go:110] failed to write ceph configuration file (open /etc/ceph/ceph.conf: permission denied)
```

The linked RedHat issue didn't fix things, but after extracting the pod definition from the deployment and running it manually all the containers started without issue.  I noticed that the deployment created pods with:

openshift.io/scc: hostaccess

Where the manually created one were running with

openshift.io/scc: hostmount-anyuid

So I added that SCC to the rook-csi-*-provisioner-sa accounts(oc adm policy add-scc-to-user hostmount-anyuid system:serviceaccount:rook-ceph:rook-csi-*-provisioner-sa ), and then killed the pods.  Once the deployment recreated them all the containers started.  The csi-rbdplugin pod still throws a warning about the PID limit but the error about writing the ceph.conf file has vanished.

Sure looks like the csi-rbdplugin pod is expecting some kind of elevated access.
